### ISSUE TITLE
WidgetSpan renders badly in the Semantics debugger

### ISSUE THREAD

**Issue by GroovinChip** [flutter/flutter#75722]
I'm exploring making an application 508-compliant for work, and noticed that `WidgetSpan` does not render well when setting `showSemanticsDebugger` to `true` in `MaterialApp`. 

## Steps to Reproduce

Toss the following code into dartpad:

```dart
import 'package:flutter/material.dart';

final Color darkBlue = Color.fromARGB(255, 18, 32, 47);

void main() {
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      theme: ThemeData.dark().copyWith(scaffoldBackgroundColor: darkBlue),
      debugShowCheckedModeBanner: false,
      home: Scaffold(
        body: Center(
          child: MyWidget(),
        ),
      ),
      showSemanticsDebugger: true,
    );
  }
}

class MyWidget extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return RichText(
      text: TextSpan(
        style: Theme.of(context).textTheme.bodyText2,
        children: [
          TextSpan(text: 'This is some text '),
          TextSpan(text: 'in a sentence made of spans '),
          WidgetSpan(
            child: Text('with a WidgetSpan '),
          ),
          TextSpan(text: 'in the middle'),
        ],
      ),
    );
  }
}
```

**Expected results:** One continuous line of text, with the `WidgetSpan` rendered in the middle

**Actual results:** The content of the `WidgetSpan` rendered below the line of text, with either an empty space or an odd OBJ box rendered where the `WidgetSpan` should be:
<img width="703" alt="Screen Shot 2021-02-09 at 2 34 26 PM" src="https://user-images.githubusercontent.com/4250470/107417769-eb231f00-6ae3-11eb-82d3-0cbbffe05576.png">

I can't attach my `flutter doctor` output because GitHub is blocked on my work computer (don't ask) so I'm filing this on my personal computer.

**Comment by darshankawar** [comment]
Hi @GroovinChip,
Tried your code sample on latest master. Commented  `showSemanticsDebugger: true` to understand the expected against actual result and I noticed below:

1. Without `showSemanticsDebugger: true`:

<img width="548" alt="Screenshot 2021-02-10 at 1 50 57 PM" src="https://user-images.githubusercontent.com/67046386/107483636-5aae1280-6ba7-11eb-9876-ed85c1c597b4.png">

2. With `showSemanticsDebugger: true`:

<img width="543" alt="Screenshot 2021-02-10 at 1 51 20 PM" src="https://user-images.githubusercontent.com/67046386/107483663-66013e00-6ba7-11eb-9608-ff26eb711e40.png">

So the expected behavior is `in the middle` text should also be shown in the `SemanticsDebugger` and without OBJ box.
Is this a correct interpretation of the issue ?

<details>
<summary> flutter doctor -v </summary>

```
[✓] Flutter (Channel master, 1.26.0-18.0.pre.255, on Mac OS X 10.15.4 19E2269 darwin-x64,
    locale en-IN)
    • Flutter version 1.26.0-18.0.pre.255 at /Users/dhs/documents/Fluttersdk/flutter
    • Framework revision 278065284d (6 hours ago), 2021-02-09 17:51:03 -0800
    • Engine revision b3a18f5182
    • Dart version 2.13.0 (build 2.13.0-15.0.dev)

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.0)
    • Android SDK at /Users/dhs/Library/Android/sdk
    • Platform android-30, build-tools 30.0.0
    • Java binary at: /Applications/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)
    • All Android licenses accepted.

[!] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.3, Build version 12C33
    ! CocoaPods 1.9.3 out of date (1.10.0 is recommended).
        CocoaPods is used to retrieve the iOS and macOS platform side's plugin code that
        responds to your plugin usage on the Dart side.
        Without CocoaPods, plugins will not work on iOS or macOS.
        For more info, see https://flutter.dev/platform-plugins
      To upgrade see https://guides.cocoapods.org/using/getting-started.html#installation
      for instructions.

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)

[✓] VS Code (version 1.53.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.18.1

[✓] Connected device (4 available)
    • SM A260G (mobile)          • 5200763ebcfa861f                     • android-arm
      • Android 8.1.0 (API 27)
    • iPhone 12 Pro Max (mobile) • A5473606-0213-4FD8-BA16-553433949729 • ios
      • com.apple.CoreSimulator.SimRuntime.iOS-14-3 (simulator)
    • macOS (desktop)            • macos                                • darwin-x64
      • Mac OS X 10.15.4 19E2269 darwin-x64
    • Chrome (web)               • chrome                               • web-javascript
      • Google Chrome 88.0.4324.150

! Doctor found issues in 1 category.

```

</details>

**Comment by GroovinChip** [comment]
> So the expected behavior is in the middle text should also be shown in the SemanticsDebugger and without OBJ box.
Is this a correct interpretation of the issue ?

Correct. It should read "This is some text in a sentence made of spans with a WidgetSpan in the middle"

**Comment by darshankawar** [comment]
Thanks for confirming. https://github.com/flutter/flutter/issues/75722#issuecomment-776532540

**Comment by olof-dev** [comment]
This is still the case with Flutter 3.0.5.

One doesn't need Material to see this; just wrap the widget in a `SemanticsDebugger` widget.

Is there an 'accessibility' github tag? If so, this issue should be tagged with it, as this issue affects people using screen readers.

**Comment by chunhtai** [comment]
This is a duplicate of https://github.com/flutter/flutter/issues/111931

**Comment by GroovinChip** [comment]
> This is a duplicate of https://github.com/flutter/flutter/issues/111931

I'd argue that #111931 is a duplicate of this issue since this one is much older.
